app/models/page.rb

```diff
@@ -37,11 +41,15 @@ def self.default_page
 
     return Page.top_level_pages.first if Page.top_level_pages.first
 
-    Page.first
+    Page.find_by( hidden: false )
```

Comment:
Consider adding an `order` to this query, `.find_by` could return different `Page`s on each call.